collections/Arch_Selden/MS_Arch_Selden_A_72_3.xml

```diff
@@ -73,7 +76,7 @@
                         <country key="place_7005560">Mexican</country>
                      </origPlace>
                   </origin>
-               </history>
+               <provenance notBefore="1584" notAfter="1654" resp="#MMM"><persName role="formerOwner" key="person_51728429">John Selden, 1584-1654</persName></provenance><acquisition when="1659" resp="#MMM">Acquired by the Bodleian in 1659</acquisition></history>
```

Comment:
The formatting could be better; this places the `provenance` element on the same level as the closing `history` element, and pushes the `history` element to the end of the line.

Reply:
I can change the script to format the XML; however, it will reformat the whole file, not just the inserted elements. See below; the `persName`, `listBibl`, `p`, and all but the most recent `change` element, which is new, have been reindented. I'm modifying an existing script that uses Nokogiri. I don't know how, in Nokogiri, to format only the inserted elements.

```diff
-<?xml version="1.0" encoding="UTF-8"?><?xml-model href="https://raw.githubusercontent.com/bodleian/consolidated-tei-schema/master/msdesc.rng" type="application/xml" schematypens
="http://relaxng.org/ns/structure/1.0"?><?xml-model href="https://raw.githubusercontent.com/bodleian/consolidated-tei-schema/master/msdesc.rng" type="application/xml" schematypen
s="http://purl.oclc.org/dsdl/schematron"?>
+<?xml version="1.0" encoding="UTF-8"?>
+<?xml-model href="https://raw.githubusercontent.com/bodleian/consolidated-tei-schema/master/msdesc.rng" type="application/xml" schematypens="http://relaxng.org/ns/structure/1.0"
?>
+<?xml-model href="https://raw.githubusercontent.com/bodleian/consolidated-tei-schema/master/msdesc.rng" type="application/xml" schematypens="http://purl.oclc.org/dsdl/schematron
"?>
 <TEI xmlns="http://www.tei-c.org/ns/1.0" xml:id="manuscript_877">
    <teiHeader>
       <fileDesc>
@@ -7,7 +9,8 @@
             <title type="collection">MSS. Barlow</title>
             <respStmt>
                <resp>Summary description</resp>
-               <persName>Elizabeth Solopova</persName><persName>Matthew Holford</persName>
+               <persName>Elizabeth Solopova</persName>
+               <persName>Matthew Holford</persName>
             </respStmt>
          </titleStmt>
          <editionStmt>
@@ -70,34 +73,37 @@
                         <country key="place_7002445">English</country>
                      </origPlace>
                   </origin>
+                  <provenance notBefore="1607" notAfter="1691" resp="#MMM">
+                     <persName role="formerOwner" key="person_465">Thomas Barlow, 1607-1691</persName>
+                  </provenance>
+                  <acquisition when="1691" resp="#MMM">Bequeathed to the Bodleian in 1691</acquisition>
                </history>
                <additional>
                   <adminInfo>
                      <recordHist>
                         <source>Summary description by Elizabeth Solopova and Matthew Holford, based on the Summary Catalogue and supplementary sources.
-                            <listBibl>
-                              <bibl type="SC" facs="aaq0408.gif">Summary Catalogue, Vol. 2 Part 2, p.  1049</bibl>
-                           </listBibl>
+                            <listBibl><bibl type="SC" facs="aaq0408.gif">Summary Catalogue, Vol. 2 Part 2, p.  1049</bibl></listBibl>
                         </source>
                      </recordHist>
                   </adminInfo>
                </additional>
             </msDesc>
          </sourceDesc>
                </history>
                <additional>
                   <adminInfo>
                      <recordHist>
                         <source>Summary description by Elizabeth Solopova and Matthew Holford, based on the Summary Catalogue and supplementary sources.
-                            <listBibl>
-                              <bibl type="SC" facs="aaq0408.gif">Summary Catalogue, Vol. 2 Part 2, p.  1049</bibl>
-                           </listBibl>
+                            <listBibl><bibl type="SC" facs="aaq0408.gif">Summary Catalogue, Vol. 2 Part 2, p.  1049</bibl></listBibl>
                         </source>
                      </recordHist>
                   </adminInfo>
                </additional>
             </msDesc>
          </sourceDesc>
-      </fileDesc>
-      <revisionDesc>          <change when="2017-07-01">First online publication.</change>
-
-         <change when="2017-05-29">
-            <persName>James Cummings</persName> Up-converted the markup using <ref target="https://github.com/jamescummings/Bodleian-msDesc-ODD/blob/master/convertTolkien2Bodley.xsl">https://github.com/jamescummings/Bodleian-msDesc-ODD/blob/master/convertTolkien2Bodley.xsl</ref>
+      </fileDesc>
+      <revisionDesc>
+         <change when="2019-11-04" xml:id="MMM"><persName>Toby Burrows/Mapping Manuscript Migrations</persName>Provenance and acquisition information added using <ref target="https://github.com/mapping-manuscript-migrations/oxford-tei-updates/blob/master/update_oxford_prov.rb">https://github.com/mapping-manuscript-migrations/oxford-tei-updates/blob/master/update_oxford_prov.rb</ref> in collaboration with the <ref target="http://mappingmanuscriptmigrations.org/">Mapping Manuscript Migrations</ref> project.</change>
+         <change when="2017-07-01">First online publication.</change>
+         <change when="2017-05-29"><persName>James Cummings</persName> Up-converted the markup using <ref target="https://github.com/jamescummings/Bodleian-msDesc-ODD/blob/master/convertTolkien2Bodley.xsl">https://github.com/jamescummings/Bodleian-msDesc-ODD/blob/master/convertTolkien2Bodley.xsl</ref>
          </change>
-         <change when="2017-05-25">
-            <persName>James Cummings</persName> Up-converted the markup using <ref target="https://github.com/jamescummings/Bodleian-msDesc-ODD/blob/master/convertTolkien2Bodley.xsl">https://github.com/jamescummings/Bodleian-msDesc-ODD/blob/master/convertTolkien2Bodley.xsl</ref>
+         <change when="2017-05-25"><persName>James Cummings</persName> Up-converted the markup using <ref target="https://github.com/jamescummings/Bodleian-msDesc-ODD/blob/master/convertTolkien2Bodley.xsl">https://github.com/jamescummings/Bodleian-msDesc-ODD/blob/master/convertTolkien2Bodley.xsl</ref>
          </change>
       </revisionDesc>
    </teiHeader>
    <text>
       <body>
-         <p><!--Body paragraph provided for validation and future transcription--></p>
+         <p>
+            <!--Body paragraph provided for validation and future transcription-->
+         </p>
       </body>
    </text>
-</TEI>
\ No newline at end of file
+</TEI>
```

Reply:
If the script being used is still similar to this one...

https://github.com/littlegustv/oxfordupdates/blob/master/test_case_for_oxford_prov.rb

...then it might improve formatting to add this...
```
doc.at('history').add_child("\n")
```
...after each of the lines that use `add_child` to add the _provenance_ and _acquisition_ to the _history_. And this...
```
doc.at('revisionDesc').prepend_child("\n")
```
...before the _change_ is prepended to the _revisionDesc_.

That should put new elements on new lines, but probably not indented.

Reply:
The script (<https://github.com/mapping-manuscript-migrations/oxford-tei-updates/blob/master/update_oxford_prov.rb>) is an updated version of the script from littlegustv. I had tried the method of inserting `"\n"` nodes. It modifies only the select elements, but doesn't result in entirely satisfactory indentation, even when telling Nokogiri to render the XML with indentation `doc.to_xml(indent: 3)`:

```diff
# snippet for collections/Barlow/MS_Barlow_12.xml
@@ -75,7 +77,9 @@
                         <country key="place_7002445">English</country>
                      </origPlace>
                   </origin>
-               </history>
+               <provenance notBefore="1607" notAfter="1691" resp="#MMM"><persName role="formerOwner" key="person_465">Thomas Barlow, 1607-1691</persName></provenance>
+<acquisition when="1691" resp="#MMM">Bequeathed to the Bodleian in 1691</acquisition>
+</history>
                <additional>
                   <adminInfo>
                      <recordHist>
@@ -91,7 +95,9 @@
             </msDesc>
          </sourceDesc>
       </fileDesc>
-      <revisionDesc>          <change when="2017-07-01">First online publication.</change>
+      <revisionDesc>
+<change when="2019-11-04" xml:id="MMM"><persName>Toby Burrows/Mapping Manuscript Migrations</persName>Provenance and acquisition information added using <ref target="https://github.com/mapping-manuscript-migrations/oxford-tei-updates/blob/master/update_oxford_prov.rb">https://github.com/mapping-manuscript-migrations/oxford-tei-updates/blob/master/update_oxford_prov.rb</ref> in collaboration with the <ref target="http://mappingmanuscriptmigrations.org/">Mapping Manuscript Migrations</ref> project.</change>
+          <change when="2017-07-01">First online publication.</change>

          <change when="2017-05-29">
```

As I understand it, the way to get libxml2 via Nokogiri, to indent the elements is to tell it to ignore whitespace when the source file is read (<https://stackoverflow.com/a/3891697>), and that results in formatting the whole document.

Reply:
@holfordm: Which would you prefer:

 * Additions inserted without line breaks
 * Each new element on its own line, but no indenting
 * Entire documents tidied up (like Oxygen's format and indent button)

Reply:
Additions without line breaks is probably the best solution. Indentation can sometimes cause problems by creating unwanted whitespace.